blob/s3blob/s3blob.go

```diff
@@ -171,6 +179,16 @@ type Options struct {
 	// Some S3-compatible services (like CEPH) do not currently support
 	// ListObjectsV2.
 	UseLegacyList bool
+
+	// EncryptionType sets the encryption type headers when making write or
+	// copy calls. This is required if the bucket has a restrictive bucket
+	// policy that enforces a specific encryption type
+	EncryptionType string
```

Comment:
Let's make this the actual type, and do the conversion from a string in the URL handler. Ideally it would return an error if it's an invalid string value at that time.